fake_svc/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"context"
+	"log"
+	"math/rand"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"go.uber.org/zap" // Import zap for structured logging
+)
+
+func initTracer() func() {
+	ctx := context.Background()
+
+	// Create OTLP trace exporter
+	exporter, err := otlptracehttp.New(ctx, otlptracehttp.WithEndpoint("http://otel-collector:4318"), otlptracehttp.WithInsecure())
+	if err != nil {
+		log.Fatalf("Failed to create trace exporter: %v", err)
+	}
+
+	// Create Resource
```

Comment:
What's a Resource?

Reply:
Turns out it's metadata for the service and other things like that. Basically to allow the tracer to identify what's using it I think?